File: library/src/main/java/io/github/anderscheow/validator/rules/common/EqualRule.java
package io.github.anderscheow.validator.rules.common;

import android.support.annotation.NonNull;
import android.support.annotation.StringRes;

import java.util.Locale;

import io.github.anderscheow.validator.rules.BaseRule;

public class EqualRule extends BaseRule {

    private String keyword;

    public EqualRule(String keyword) {
        super(String.format(Locale.getDefault(), "Value does not equal to '%s'", keyword));
        this.keyword = keyword;
    }

    public EqualRule(String keyword, @StringRes int errorRes) {
        super(errorRes);
        this.keyword = keyword;
    }

    public EqualRule(String keyword, @NonNull String errorMessage) {
        super(errorMessage);
        this.keyword = keyword;
    }

    @Override
    public boolean validate(String value) {
        return value.equals(keyword);
    }
}

File: library/src/main/java/io/github/anderscheow/validator/rules/common/EqualRule.kt
package io.github.anderscheow.validator.rules.common

import androidx.annotation.StringRes
import io.github.anderscheow.validator.rules.Rule

class EqualRule : Rule {

    private var keyword: Any

    constructor(keyword: Any, @StringRes errorRes: Int) :
            super(errorRes) {
        this.keyword = keyword
    }

    constructor(keyword: Any, errorMessage: String) :
            super(errorMessage) {
        this.keyword = keyword
    }

    override fun validate(value: String?): Boolean {
        if (value == null) {
            throw NullPointerException()
        } else {
            return value == keyword
        }
    }
}

fun equalTo(keyword: String, @StringRes errorRes: Int): EqualRule = EqualRule(keyword, errorRes)

fun equalTo(keyword: String, errorMessage: String): EqualRule = EqualRule(keyword, errorMessage)